README
NAME
    Class::Value::Net - Network-related value objects

VERSION
    version 1.103070

INSTALLATION
    See perlmodinstall for information and options on installing Perl
    modules.

BUGS AND LIMITATIONS
    No bugs have been reported.

    Please report any bugs or feature requests through the web interface at
    <http://rt.cpan.org/Public/Dist/Display.html?Name=Class-Value-Net>.

AVAILABILITY
    The latest version of this module is available from the Comprehensive
    Perl Archive Network (CPAN). Visit <http://www.perl.com/CPAN/> to find a
    CPAN site near you, or see
    <http://search.cpan.org/dist/Class-Value-Net/>.

    The development version lives at
    <http://github.com/hanekomu/Class-Value-Net> and may be cloned from
    <git://github.com/hanekomu/Class-Value-Net>. Instead of sending patches,
    please fork this project using the standard git and github
    infrastructure.

AUTHOR
    Marcel Gruenauer <marcel@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2004 by Marcel Gruenauer.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Perl extension Class-Value-Net

1.103070  2010-11-03 12:31:06 Europe/Vienna
    - implemented DNSSEC RR DS

1.103060  2010-11-02 15:31:30 Europe/Vienna

1.101870  2010-07-06 13:39:50 Europe/Vienna
    - use Data::Semantic::Net::IPAddress::TestData::IPv4 1.101750

1.101730  2010-06-22 15:00:24 Europe/Vienna
    - added minimum version for ::TestData::IPv4

1.101690  2010-06-18 13:04:17 Europe/Vienna
    - Class::Value::Net::IPAddress: forbid_internal defaults to 1 now

1.101610  2010-06-10 21:53:36 Europe/Vienna
    - Class::Value::Net::IPAddress: use Error::Hierarchy::Mixin to get record()

1.101370  2010-05-17 17:45:08 Europe/Vienna
    - local $Class::Value::SkipChecks = 0; # where necessary

1.100840  2010-03-25 20:16:13 Europe/Vienna
    - converted the distribution to Dist::Zilla-style

0.05  Thu Jul 25 15:27:43 CEST 2008 (Marcel Gruenauer <marcel@cpan.org>)
     - Changed dist style and Changes back to standard. People didn't like it -
       the nail that sticks out gets hammered down.
     - Added standard test files; this will also help with CPANTS.

0.04  Thu, 29 May 2008 12:02:16 -0000 (Marcel Gruenauer <marcel@cpan.org>)
     - set the version to 0.04
     - moved generated documentation of interited methods to after
       documentation of this class' methods so the manpage is more readable
     - .shipit: fixed svk.tagpattern

0.03  Thu, 22 Nov 2007 13:08:28 -0000 (Marcel Gruenauer <marcel@cpan.org>)
     - set the version to 0.03
     - Makefile.PL: now requires Test::CompanionClasses 0.03; earlier versions
       had a bug in make_real_object()

0.02  Wed, 21 Nov 2007 18:21:27 -0000 (Marcel Gruenauer <marcel@cpan.org>)
     - Converted Changes file to YAML style
     - .shipit: added Twitter step
     - Makefile.PL: added auto_install() and process_templates()
     - lib/*: converted to template
     - updated MANIFEST
     - slightly improved documentation
     - Class::Value::Net::IPAddress::IPv4 and ::IPv6 now subclass
       Class::Value::SemanticAdapter
     - added lib/Class/Value/Net/IPAddress/IPv4_TEST.pm
     - deleted t/README.txt
     - Makefile.PL: require Class::Value::SemanticAdapter, Data::Semantic::Net
     - Class::Value::Net::IPAddress::IPv4, ::IPv6: inherit from
       Class::Value::Net::IPAddress, moved is_internal() there
     - Class::Value::Net::IPAddress::IPv4_TEST: just inherit from
       Class::Value::Test and Data::Semantic::Net::IPAddress::TestData::IPv4
     - Class::Value::Net::IPAddress: inherit from
       Class::Value::SemanticAdapter, remove is_valid_normalized_value(), add
       semantic_args(), add is_internal()
     - Class::Value::Net::NAPTR::CharacterString_TEST: adapted to TESTDATA
       style
     - Class::Value::Net::NAPTR::Flags_TEST: adapted to TESTDATA style
     - Class::Value::Net: reflow
     - updated MANIFEST
     - added lib/Class/Value/Net/IPAddress/IPv6_TEST.pm
     - getting rid of the distinction between well-formedness and validity.
       Only validity remains
     - Class::Value::Net::IPAddress::IPv4, ::IPv6: replaced pod
     - updated t/perlcriticrc
     - tags: NEWFEATURE

0.01  Wed, 07 Nov 2007 09:59:46 +0100 (Marcel Gruenauer <marcel@cpan.org>)
     - original version


Makefile.PL

use strict;
use warnings;

BEGIN { require 5.008; }

use ExtUtils::MakeMaker 6.31;



my %WriteMakefileArgs = (
  'ABSTRACT' => 'Network-related value objects',
  'AUTHOR' => 'Marcel Gruenauer <marcel@cpan.org>',
  'BUILD_REQUIRES' => {
    'Carp' => '0',
    'Class::Value' => '0',
    'English' => '0',
    'Error' => '0',
    'File::Find' => '0',
    'File::Temp' => '0',
    'Scalar::Util' => '0',
    'Test::CompanionClasses' => '0',
    'Test::More' => '0.94'
  },
  'CONFIGURE_REQUIRES' => {
    'ExtUtils::MakeMaker' => '6.31'
  },
  'DISTNAME' => 'Class-Value-Net',
  'EXE_FILES' => [],
  'LICENSE' => 'perl',
  'NAME' => 'Class::Value::Net',
  'PREREQ_PM' => {
    'Class::Value::Boolean' => '0',
    'Class::Value::Enum' => '0',
    'Class::Value::Exception::Base' => '0',
    'Class::Value::SemanticAdapter' => '0',
    'Class::Value::String' => '0',
    'Class::Value::Test' => '0',
    'Data::Semantic::Net::IPAddress::TestData::IPv4' => '1.101750',
    'Data::Semantic::Net::IPAddress::TestData::IPv6' => '0',
    'Error::Hierarchy::Mixin' => '0',
    'Error::Hierarchy::Test' => '0',
    'Test::CompanionClasses::Base' => '0',
    'Test::More' => '0',
    'constant' => '0',
    'parent' => '0'
  },
  'VERSION' => '1.103070',
  'test' => {
    'TESTS' => 't/*.t'
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



